src/legilogic_ethereum/ethereum_rlp.ml

```diff
@@ -48,6 +48,12 @@ let decode_int_string n =
 let decode (RlpEncoding s) =
   Rlp_decode.rlp_item_of_rlp s
 
+let amount_or_zero : TokenAmount.t option -> TokenAmount.t =
+  fun val_opt ->
+  match val_opt with
```

Comment:
Do we need it? If we do it's just `Option.default TokenAmount.zero` and probably doesn't deserve a name.

Reply:
Actually, the `ppx_yojson` library allows you to specify the default when reading from JSON.